pkg/extensions/search/resolver.go

```diff
@@ -74,19 +72,14 @@ func NewResolver(log log.Logger, storeController storage.StoreController,
 	return resolver
 }
 
-func FilterByDigest(digest string) mTypes.FilterFunc {
-	return func(repoMeta mTypes.RepoMetadata, manifestMeta mTypes.ManifestMetadata) bool {
+func ProtoFilterByDigest(digest string) mTypes.FilterFunc {
+	return func(repoMeta mTypes.RepoMetadata, imageData mTypes.ImageData) bool {
 		lookupDigest := digest
 		contains := false
 
-		var manifest ispec.Manifest
+		manifest := imageData.Manifests[0]
```

Comment:
Is it safe to use just the first manifest? We need to check any manifests, I assume the old code worked for indexes too.

Reply:
This is applied on the manifest level, the old functions worked the same way. We can't have an image without at least 1 manifest